pkg/es/config/config.go

```diff
@@ -70,34 +81,56 @@ func (p IndexPrefix) Apply(indexName string) string {
-	BulkActions              int                    `mapstructure:"-"`
-	BulkFlushInterval        time.Duration          `mapstructure:"-"`
-	Indices                  Indices                `mapstructure:"indices"`
-	ServiceCacheTTL          time.Duration          `mapstructure:"service_cache_ttl"`
-	AdaptiveSamplingLookback time.Duration          `mapstructure:"-"`
-	Tags                     TagsAsFields           `mapstructure:"tags_as_fields"`
-	Enabled                  bool                   `mapstructure:"-"`
-	TLS                      configtls.ClientConfig `mapstructure:"tls"`
-	UseReadWriteAliases      bool                   `mapstructure:"use_aliases"`
-	CreateIndexTemplates     bool                   `mapstructure:"create_mappings"`
-	UseILM                   bool                   `mapstructure:"use_ilm"`
-	Version                  uint                   `mapstructure:"version"`
-	LogLevel                 string                 `mapstructure:"log_level"`
-	SendGetBodyAs            string                 `mapstructure:"send_get_body_as"`
+	// Servers is a list of Elasticsearch servers. The strings must must contain full URLs
+	// (i.e. http://localhost:9200).
+	Servers []string `mapstructure:"server_urls" valid:"required,url"`
+	// RemoteReadClusters is a list of Elasticsearch remote cluster names for cross-cluster
+	// querying.
+	RemoteReadClusters []string       `mapstructure:"remote_read_clusters"`
+	Authentication     Authentication `mapstructure:"auth"`
+	// TokenFilePath contains the path to a file containing a bearer token.
+	// This flag also loads CA if it is specified.
+	TokenFilePath string `mapstructure:"token_file"`
+	// AllowTokenFromContext, if set to true, enables reading bearer token from the context.
+	AllowTokenFromContext bool     `mapstructure:"allow_token_from_context"`
+	Sniffing              Sniffing `mapstructure:"sniffing"`
+	// MaxDocCount Defines maximum number of results to fetch from storage per query.
+	MaxDocCount int `mapstructure:"max_doc_count"`
+	// MaxSpanAge configures the maximum lookback on span reads.
+	MaxSpanAge time.Duration `mapstructure:"max_span_age"`
+	// Timeout contains the timeout used for queries. A timeout of zero means no timeout.
+	Timeout        time.Duration  `mapstructure:"timeout"`
+	BulkProcessing BulkProcessing `mapstructure:"bulk_processing"`
+	Indices        Indices        `mapstructure:"indices"`
+	// ServiceCacheTTL contains the TTL for the cache of known service names
+	ServiceCacheTTL time.Duration `mapstructure:"service_cache_ttl"`
+	// AdaptiveSamplingLookback contains the duration to look back for the
+	// latest adaptive sampling probabilities.
+	AdaptiveSamplingLookback time.Duration `mapstructure:"adaptive_sampling_lookback"`
+	Tags                     TagsAsFields  `mapstructure:"tags_as_fields"`
+	// Enabled, if set to true, enables the namespace for storage pointed to by this configuration.
+	Enabled bool `mapstructure:"enable"`
```

Comment:
This is very confusing. Do we even need it?

Reply:
I agree but it looks to exist to explicitly enable archive storage
- https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/es/factory.go#L148
- https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/es/options.go#L109-L110